File: NumberCount/numberCount.cpp
//number count using maps data structure
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        std::map<int,int> M;
        for (int i{0};i<nums.size(); i++)
        {
            if(M.find(nums[i]) == M.end()) M[nums[i]] = 1;
            else M[nums[i]] +=1;
        }
        for (auto& it : M)
        {
            if (it.second == 1)
                return it.first;
        }
        return 1;
    }
};

File: NumberCount/numbercount.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        for i in nums:
            if nums.count(i)== 1:
                return i


#or we can use counter like so with this being much faster than the above method
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        from collections import Counter
        a = Counter(nums)
        key = [k for k, v in a.items() if v == 1][0]
        #the use of [0] is to select the value from the returned list as the above method returns a list
        return key
